ghost/milestone-emails/lib/MilestonesEmailService.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * @typedef {import('./MilestonesAPI')} MilestonesAPI
+ * @typedef {import('./Milestone')} Milestone
+ */
+
+/**
+ * @template Model
+ * @typedef {import('./MilestonesAPI')} <Model>
+ */
+
+/**
+ * @typedef {Object} IQueries
+ * @prop {() => Promise<number>} getMembersCount
+ * @prop {() => Promise<Object>} getARR
+ * @prop {() => Promise<boolean>} hasImportedMembersInPeriod
+ */
+
+module.exports = class MilestonesEmailService {
+    /** @type {import('./MilestonesAPI')} */
+    #api;
+
+    /** @type {Function} */
+    #mailer;
+
+    /** @type {Object} */
+    #config;
+
+    /** @type {IQueries} */
+    #queries;
+
+    /** @type {string} */
+    #defaultCurrency;
+
+    /**
+     * @param {object} deps
+     * @param {Function} deps.mailer
+     * @param {MilestonesAPI} deps.api
+     * @param {Object} deps.config
+     * @param {IQueries} deps.queries
+     * @param {string} deps.defaultCurrency
+     */
+    constructor(deps) {
+        this.#mailer = deps.mailer;
+        this.#api = deps.api;
+        this.#config = deps.config;
+        this.#queries = deps.queries;
+        this.#defaultCurrency = deps.defaultCurrency;
```

Comment:
@allouis after checking, we'll have to expect to only deal with a default currency. Probably the currency as defined in the default product. Still need to figure out on the other end to grab that.